diff --git a/dist/Repl.js b/dist/Repl.js
index 75b69e75c566731d30d92699ff17214b4cf314d9..9b2a15aab186a6661471cec7f5ac5ce7a5eaa229 100644
--- a/dist/Repl.js
+++ b/dist/Repl.js
@@ -675,7 +675,8 @@ function Repl({
   );
   const extractStyles = () => {
     var _a, _b;
-    const cmEditorEl = (_b = (_a = rcm.current) == null ? void 0 : _a.editor) == null ? void 0 : _b.querySelector(".cm-editor");
+    const cmEditorEl = document.querySelector(".cm-editor")
+    // const cmEditorEl = (_b = (_a = rcm.current) == null ? void 0 : _a.editor) == null ? void 0 : _b.querySelector(".cm-editor");
     if (!cmEditorEl) {
       throw new Error("No CodeMirror editor found");
     }
